react/components/BuyButton/index.js

```diff
@@ -45,44 +42,60 @@ export class BuyButton extends Component {
   }
 
   handleAddToCart = async () => {
-    const { skuItems, isOneClickBuy, orderFormContext, push } = this.props
+    const { addToCart, skuItems, isOneClickBuy } = this.props
     this.setState({ isAddingToCart: true })
 
-    const variables = {
-      items: skuItems.map(skuItem => {
-        const { skuId } = skuItem
+    if (isOneClickBuy) location.assign(CONSTANTS.CHECKOUT_URL)
```

Comment:
Why before the action? And if something went wrong?

Reply:
I don't know how it was working before. And back then, it was before the action too...